front/pages/w/[wId]/subscription/awaiting_confirmation.tsx

```diff
@@ -0,0 +1,112 @@
+import { BarHeader, Page, Spinner2 } from "@dust-tt/sparkle";
+import type { UserType, WorkspaceType } from "@dust-tt/types";
+import type { SubscriptionType } from "@dust-tt/types";
+import type { InferGetServerSidePropsType } from "next";
+import { useRouter } from "next/router";
+import React, { useEffect, useState } from "react";
+
+import { PRO_PLAN_29_COST } from "@app/lib/client/subscription";
+import { withDefaultUserAuthRequirements } from "@app/lib/iam/session";
+import { getStripeSubscription } from "@app/lib/plans/stripe";
+import { countActiveSeatsInWorkspace } from "@app/lib/plans/usage/seats";
+
+const { GA_TRACKING_ID = "" } = process.env;
+
+export const getServerSideProps = withDefaultUserAuthRequirements<{
+  owner: WorkspaceType;
+  subscription: SubscriptionType;
+  user: UserType;
+  trialDaysRemaining: number | null;
+  gaTrackingId: string;
+  workspaceSeats: number;
+  estimatedMonthlyBilling: number;
+}>(async (context, auth) => {
```

Comment:
Would be handled by the front end, agree it could be done on the backend side too but this adds complexity to the code with very little other upside so I am keeping it that way.

Reply:
Where is it being done in the Front end?

In case you want to add it:
```
if (subscription.plan.limits.canUseProduct) {
  return {
    redirect: {
      permanent: false,
      destination: `/w/${context.query.wId}/assistant/new`,
    },
  };
}
```

Reply:
@flvndvd I initially added it, but this is actually not the right redirect, we should redirect on the `/congratulation` page.

I understand this redirect makes sense from a pure technical perspective and I agree with it, but from a practical perspective, this put this user flow at higher risk of not being right and I just spent 15 minutes trying to understand why it was buggy again. It does not add much to the user experience and cost us very little, so I will remove these lines and let the front-end do the work. 
I hope you are fine with it 🙏